src/borg/archiver.py

```diff
@@ -3338,6 +3339,9 @@ def get_args(self, argv, cmd):
         if cmd is not None and result.func == self.do_serve:
             forced_result = result
             argv = shlex.split(cmd)
+            # Drop BORG_ environment variables (do *not* interpret them) before trying to parse
+            # the borg command line.
+            argv = list(itertools.dropwhile(lambda arg: arg.startswith('BORG_'), argv))
```

Comment:
well, we (currently) only put BORG_* env vars there, but isn't the rule rather that we need to drop all env vars, because having FOO=BAR instead of 'borg' as argv[0] just shifts everything to the right and confuses the arg parser?

So, just use `'=' in arg` as dropwhile condition?